client/app/views/contacts/index.coffee

```diff
@@ -63,65 +73,97 @@ module.exports = class Contacts extends Mn.CompositeView
         return elBuffer
 
 
-    filter: (model, index, collection) ->
-        return true unless model instanceof ContactViewModel
-
-        app = require 'application'
-
-        if @tag and @tag not in model.get('tags')
-            return false
-
-        if app.model.get 'scored'
-            model.model.cid in Object.keys @collection.filtered
-        else
-            char    = @model.get 'name'
-            idx     = if app.model.get('sort') is 'fn' then 0 else 1
-            initial = model.get('initials')[idx]
-
-            if char is '#' then not initial?.match(/[a-zA-Z]/)
-            else char is initial?.toLowerCase()
+    # filter: (model, index, collection) ->
+    #     return true unless model instanceof ContactViewModel
+    #
+    #     app = require 'application'
+    #
+    #     if @tag and @tag not in model.get('tags')
+    #         return false
+    #
+    #     if app.model.get 'scored'
+    #         model.model.cid in Object.keys @collection.filtered
+    #     else
+    #         char    = @model.get 'name'
+    #         idx     = if app.model.get('sort') is 'fn' then 0 else 1
+    #         initial = model.get('initials')[idx]
+    #
+    #         if char is '#' then not initial?.match(/[a-zA-Z]/)
+    #         else char is initial?.toLowerCase()
 
 
     initialize: ->
         app = require 'application'
 
-        if @model?._attachedCollection
-            @collection = @model._attachedCollection
+        @indexed = !!(@model or app.model.get 'scored')
+
+        if @model?.has 'char'
+            @collection = app.filtered
+            @hasContacts = true
+
+            @filter = (vmodel) =>
+                _.contains @collection.indexes.get(@model.get('char')), vmodel
 
-        else unless @model or app.model.get 'scored'
-            collection = @collection
-            groups = [].map.call 'abcdefghijklmnopqrstuvwxyz#', (char) ->
-                group = new Backbone.Model name: char
-                group._attachedCollection = collection
-                return group
-            @collection = new Backbone.Collection groups
+            @listenTo @collection,
+                'update': ->
+                    @updateTagClasses()
+                    # @toggleEmpty()
+
+            @listenTo @collection.underlying,
+                'change:tags': ->
+                    @updateTagClasses()
+                # 'add': ->
+                #     @toggleEmpty()
+                # 'remove': ->
+                #     @toggleEmpty()
+
+        else unless @indexed
+            @collection = new Backbone.Collection [].map.call indexes, (char) ->
```

Comment:
> Let's say Array.protoype.map

Let's go!

> Just for information, what's the point to call a map function on a empty array?

`indexes` contains a string composed of all `[a-z]` letters + `#`. We need to build a `Map` in which keys are one letter, associated to an array that'll contain the attached ViewModels (for the corresponding key initial).

Calling Array/prototype.map on `indexes` string'll iterate on each character and return an array containing the key and an empty array (e.g. `[ key, [] ]`). It'll be used to instanciate the `Map` [constructor](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Map#Relation_with_Array_objects).